workers/unity/Packages/io.improbable.gdk.debug/NetStatsViewer/NetStatsTab.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core.NetworkStats;
+using UnityEditor;
+using UnityEngine.UIElements;
+
+namespace Improbable.Gdk.Debug.NetStats
+{
+    internal abstract class NetStatsTab : VisualElement
+    {
+        private const string UpdateRowUxmlPath =
+            "Packages/io.improbable.gdk.debug/NetStatsViewer/Templates/UpdateRow.uxml";
+
+        private NetworkStatisticsSystem netStatSystem;
+        protected List<(string, MessageTypeUnion)> ElementInfo;
+
+        protected abstract void PopulateElementInfo();
+
+        public void InitializeTab()
+        {
+            PopulateElementInfo();
+
+            // Load update row
+            var itemTemplate = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(UpdateRowUxmlPath);
+
+            // Clone row for each component, fill list
+            var scrollView = this.Q<ScrollView>("container");
+            for (var i = 0; i < ElementInfo.Count; i++)
+            {
+                var element = itemTemplate.CloneTree();
+                UpdateElement(i, element);
+                scrollView.Add(element);
+            }
+        }
+
+        public void Update()
+        {
+            var scrollView = this.Q<ScrollView>("container");
+            var count = scrollView.childCount;
+            var viewportBound = scrollView.contentViewport.localBound;
+
+            // Update all visible items in the list
+            for (var i = 0; i < count; i++)
+            {
+                var element = scrollView[i];
+                if (viewportBound.Overlaps(element.localBound))
+                {
+                    UpdateElement(i, element);
+                }
+            }
+        }
+
+        internal void SetSystem(NetworkStatisticsSystem system)
+        {
+            netStatSystem = system;
+        }
+
+        // TODO: Temporary placeholder to allow sorting
+        private (string name, MessageTypeUnion messageType) GetInfoForIndex(int index)
+        {
+            return ElementInfo[index];
+        }
+
+        private void UpdateElement(int index, VisualElement element)
+        {
+            // Set component name
+            var (elementName, messageType) = GetInfoForIndex(index);
+            var nameElement = element.Q<Label>("name");
+            nameElement.text = elementName;
+            nameElement.tooltip = elementName;
+
+            if (netStatSystem?.World == null || !netStatSystem.World.IsCreated)
+            {
+                return;
+            }
+
+            // Incoming stats
+            var (dataIn, time) =
+                netStatSystem.GetSummary(messageType, 60, Direction.Incoming);
+
+            if (time == 0)
```

Comment:
what is this xD
reading the next line it makes sense, but this needs a comment

Reply:
This is avoiding a division by 0 when there are no samples

Reply:
as said: I understand why, but would be good to have it in code so you know already why it's gonna happen

Reply:
Will do